packages/lib/src/vis/heatmap/utils.ts

```diff
@@ -111,3 +151,15 @@ export function getInterpolator(colorMap: ColorMap, reverse: boolean) {
   const interpolator = INTERPOLATORS[colorMap];
   return reverse ? (t: number) => interpolator(1 - t) : interpolator;
 }
+
+export function getTextureFormatFromType(type: TextureDataType): PixelFormat {
+  if (type === FloatType || type === HalfFloatType) {
+    return RedFormat;
+  }
+
+  if (UNSUPPORTED_TEXTURE_TYPES.has(type)) {
+    throw new Error('Texture type not supported');
+  }
```

Comment:
TS won't let me create a narrower version of `TextureDataType` with the unsupported types excluded... I could only manage to do a runtime check...

Reply:
Unfortunate but that will work too